### DISCUSSION TITLE
The .pt model file cannot be found after converting the spec file to exe file.

### DISCUSSION THREAD

**Discussion by CCccccccc98** [ultralytics/ultralytics#8320] [Q&A]
I want to convert my .py file into exe. I originally encountered an error that the default.yaml file could not be found. Later, after searching online, I created a main.spec file and then added it to it. In the following paragraph,
![螢幕擷取畫面 2024-02-20 154522](https://github.com/ultralytics/ultralytics/assets/160203768/e37878de-5172-4631-b779-2420c4f93c19)
 the default.yaml in the ultralytics file can be found, but then a new problem occurred to me. I could not find the .pt file that the original py file introduced into my module. I added this sentence ultra_files.append( (r'C:\Users\HP\Desktop\archive\playingCards.pt', 'ultralytics'))
But it still can’t be found. Below is my main folder archive, which contains the original .py file and my model .pt file. There is also an ultralytics folder, as well as the errors I encountered. I would like to ask everyone for help. Thanks
![螢幕擷取畫面 2024-02-20 153153](https://github.com/ultralytics/ultralytics/assets/160203768/b3608a6c-7691-43de-8c57-dcbe50161926)
![螢幕擷取畫面 2024-02-20 153045](https://github.com/ultralytics/ultralytics/assets/160203768/6a21c8f8-0ec0-44b7-9722-78bbb67a6dd3)

![螢幕擷取畫面 2024-02-20 153204](https://github.com/ultralytics/ultralytics/assets/160203768/37d9061a-f213-4ac2-ab2e-790dce5f6e08)

**Comment by glenn-jocher** [comment]
Hey there! 👋 It looks like you're having trouble bundling the `.pt` model file with your executable. When using PyInstaller, you need to ensure that all required files are specified in the `.spec` file so they can be included in the final build.

Here's a snippet you can add to your `.spec` file to include your `.pt` model file:

```python
a = Analysis(['your_script.py'],
             ...
             datas=[('path/to/your/model.pt', '.')],
             ...)
```

Make sure to replace `'path/to/your/model.pt'` with the actual path to your `.pt` file, and `'your_script.py'` with the name of your Python script. The `'.'` indicates that the file should be placed in the root directory of the executable.

After updating the `.spec` file, rebuild your executable with PyInstaller:

```bash
pyinstaller your_main.spec
```

Give this a try, and let us know if it resolves the issue! If you're still facing problems, we're here to help. 😊

**Reply by CCccccccc98** [reply]
I finally used auto-py-to-exe to drop all the files and folders to be used, and successfully solved the problem.

**Reply by glenn-jocher** [reply]
@CCccccccc98 that's great to hear! 🎉 If you encounter any more issues or have further questions, feel free to reach out. Happy coding with YOLOv8!
